x-pack/packages/kbn-elastic-assistant/impl/assistant/api/conversations/use_bulk_actions_conversations.ts

```diff
@@ -0,0 +1,152 @@
+}
+
+export interface ConversationUpdateParams {
+  id?: string;
+  title?: string;
+  messages?: Message[];
+  apiConfig?: {
+    connectorId?: string;
+    connectorTypeTitle?: string;
+    defaultSystemPromptId?: string;
+    provider?: OpenAiProviderType;
+    model?: string;
+  };
+}
+
+export interface ConversationsBulkActions {
+  update?: Record<string, ConversationUpdateParams>;
+  create?: Record<string, Conversation>;
+  delete?: {
+    ids: string[];
+  };
+}
+
+const transformCreateActions = (
+  createActions: Record<string, Conversation>,
+  conversationIdsToDelete?: string[]
+) =>
+  Object.keys(createActions).reduce((conversationsToCreate: Conversation[], conversationId) => {
+    if (createActions && !conversationIdsToDelete?.includes(conversationId)) {
+      conversationsToCreate.push(createActions[conversationId]);
+    }
+    return conversationsToCreate;
+  }, []);
+
+const transformUpdateActions = (
+  updateActions: Record<string, ConversationUpdateParams>,
+  conversationIdsToDelete?: string[]
+) =>
+  Object.keys(updateActions).reduce(
+    (conversationsToUpdate: ConversationUpdateParams[], conversationId) => {
+      if (updateActions && !conversationIdsToDelete?.includes(conversationId)) {
+        conversationsToUpdate.push({
+          id: conversationId,
+          ...updateActions[conversationId],
+        });
+      }
+      return conversationsToUpdate;
+    },
+    []
+  );
+
+export const bulkChangeConversations = async (
+  http: HttpSetup,
+  conversationsActions: ConversationsBulkActions,
+  toasts?: IToasts
+) => {
+  // transform conversations disctionary to array of Conversations to create
+  // filter marked as deleted
+  const conversationsToCreate = conversationsActions.create
+    ? transformCreateActions(conversationsActions.create, conversationsActions.delete?.ids)
+    : undefined;
+
+  // transform conversations disctionary to array of Conversations to update
+  // filter marked as deleted
+  const conversationsToUpdate = conversationsActions.update
+    ? transformUpdateActions(conversationsActions.update, conversationsActions.delete?.ids)
+    : undefined;
+
+  try {
+    const result = await http.fetch<BulkActionResponse>(
+      ELASTIC_AI_ASSISTANT_CONVERSATIONS_URL_BULK_ACTION,
+      {
+        method: 'POST',
+        version: ELASTIC_AI_ASSISTANT_API_CURRENT_VERSION,
+        body: JSON.stringify({
+          update: conversationsToUpdate,
+          create: conversationsToCreate,
+          delete: conversationsActions.delete,
+        }),
+      }
+    );
+
+    if (!result.success) {
+      const serverError = result.attributes.errors
+        ?.map(
+          (e) =>
+            `${e.status_code ? `Error code: ${e.status_code}. ` : ''}Error message: ${
+              e.message
+            } for conversation ${e.conversations.map((c) => c.name).join(',')}`
+        )
+        .join(',\n');
+      throw new Error(serverError);
+    }
+    return result;
```

Comment:
nit: When mutating conversations we can update the local cache from the response instead of re-fetching. Right now when changing the connector on a conversation I'm seeing a PUT followed immediately by a GET for fetching the latest:

<img width="1360" alt="image" src="https://github.com/elastic/kibana/assets/2946766/63dfe38b-6d47-4d90-bdbc-b1523dcc4d40">



This is the pattern we used over on the rule management side to update the existing rule cache when a single rule has been mutated:

https://github.com/elastic/kibana/blob/3a18916b31d27d2ffeb7df682e69224a273b39b4/x-pack/plugins/security_solution/public/detection_engine/rule_management/api/hooks/use_fetch_rule_by_id_query.ts#L63-L85